src/Core/MailTemplate/MailTemplateFolderCatalog.php

```diff
@@ -0,0 +1,191 @@
+ */
+
+namespace PrestaShop\PrestaShop\Core\MailTemplate;
+
+use PrestaShop\PrestaShop\Core\Exception\InvalidException;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+/**
+ * This is a basic mail templates catalog, not a lot of intelligence it is based
+ * simply on existing files on the $mailThemesFolder (no database, or config files).
+ */
+class MailTemplateFolderCatalog implements MailTemplateCatalogInterface
+{
+    /** @var string */
+    private $mailThemesFolder;
+
+    /**
+     * MailTemplateFolderCatalog constructor.
+     *
+     * @param string $mailThemesFolder
+     */
+    public function __construct($mailThemesFolder)
+    {
+        $this->mailThemesFolder = $mailThemesFolder;
+    }
+
+    /**
+     * Returns the list of found themes (non empty folders, in the mail themes
+     * folder).
+     *
+     * @throws InvalidException
+     *
+     * @return string[]
+     */
+    public function listThemes()
+    {
+        $this->checkTemplatesFolder();
+
+        $finder = new Finder();
+        $finder->directories()->in($this->mailThemesFolder)->depth(0);
+        $themes = [];
+        /** @var SplFileInfo $themeFolder */
+        foreach ($finder as $themeFolder) {
+            $dirFinder = new Finder();
+            $dirFinder->files()->in($themeFolder->getRealPath());
+            if ($dirFinder->count() > 0) {
+                $themes[] = $themeFolder->getFilename();
+            }
+        }
+
+        return $themes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function listTemplates($theme)
+    {
+        $this->checkTemplatesFolder();
+
+        $collection = new MailTemplateCollection();
+        $this->listCoreTemplates($collection, $theme);
+        $this->listModulesTemplates($collection, $theme);
+
+        return $collection;
+    }
+
+    /**
+     * @param MailTemplateCollectionInterface $collection
+     * @param string $theme
+     */
+    private function listCoreTemplates(MailTemplateCollectionInterface $collection, $theme)
+    {
+        $coreTemplatesFolder = implode(DIRECTORY_SEPARATOR, [
+            $this->mailThemesFolder,
+            $theme,
+            MailTemplateInterface::CORE_CATEGORY,
+        ]);
+        if (!file_exists($coreTemplatesFolder) || !is_dir($coreTemplatesFolder)) {
+            return;
+        }
+
+        $this->addTemplatesFromFolder($collection, $coreTemplatesFolder, $theme, MailTemplateInterface::CORE_CATEGORY);
+    }
+
+    /**
+     * @param MailTemplateCollectionInterface $collection
+     * @param string $theme
+     */
+    private function listModulesTemplates(MailTemplateCollectionInterface $collection, $theme)
+    {
+        $moduleTemplatesFolder = implode(DIRECTORY_SEPARATOR, [
+            $this->mailThemesFolder,
+            $theme,
+            MailTemplateInterface::MODULES_CATEGORY,
+        ]);
+        if (!file_exists($moduleTemplatesFolder) || !is_dir($moduleTemplatesFolder)) {
```

Comment:
Same

Reply:
And the check is the same a previous, maybe use one function to prevent duplicate.